Composer/packages/server/src/services/project.ts

```diff
@@ -435,6 +442,131 @@ export class BotProjectService {
     }
   };
 
+  public static async migrateProjectAsync(req: Request, jobId: string) {
+    const { oldProjectId, name, description, location, storageId } = req.body;
+    const user = await ExtensionContext.getUserFromRequest(req);
+
+    try {
+      const locationRef = getLocationRef(location, storageId, name);
+
+      await BotProjectService.cleanProject(locationRef);
+
+      // Update status for polling
+      BackgroundProcessManager.updateProcess(jobId, 202, formatMessage('Getting template'));
+
+      const newProjRef = await AssetService.manager.copyRemoteProjectTemplateToV2(
+        '@microsoft/generator-microsoft-bot-adaptive',
+        '*', // use any available version
+        name,
+        locationRef,
+        jobId,
+        'webapp',
+        'dotnet',
+        user
+      );
+
+      // update project ref to point at newly created folder
+      newProjRef.path = `${newProjRef.path}/${name}`;
+
+      BackgroundProcessManager.updateProcess(jobId, 202, formatMessage('Migrating data'));
+
+      const originalProject = await BotProjectService.getProjectById(oldProjectId, user);
+      if (originalProject.settings) {
+        const originalFiles = originalProject.getProject().files;
+
+        // pass in allowPartialBots = true so that this project can be opened even though
+        // it doesn't yet have a root dialog...
+        const id = await BotProjectService.openProject(newProjRef, user, { allowPartialBots: true });
+        const currentProject = await BotProjectService.getProjectById(id, user);
+
+        // add all original files to new project
+        for (let f = 0; f < originalFiles.length; f++) {
+          await currentProject.migrateFile(
+            originalFiles[f].name,
+            originalFiles[f].content,
+            originalProject.rootDialogId
+          );
+        }
+
+        const newSettings = {
+          ...currentProject.settings,
+          runtimeSettings: {
+            features: {
+              showTyping: originalProject.settings?.feature?.UseShowTypingMiddleware || false,
+              useInspection: originalProject.settings?.feature?.UseInspectionMiddleware || false,
+              removeRecipientMentions: originalProject.settings?.feature?.RemoveRecipientMention || false,
+              setSpeak: originalProject.settings?.feature?.useSetSpeakMiddleware || false,
```

Comment:
This one was added by @tracyboehrer and it is now an object instead of a boolean. @tracyboehrer could you help us get this one right?

I think it will look something like (pseudo logic, Tracy should help us with the precise one): `setSpeak: { }` based on the properties here: https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime/Settings/SpeakSettings.cs

So the presence of the object enables setSpeak. I don't know what are the sensible default values to enter when we get, from the migrated file, a 'true'.